UIforETW/UIforETWDlg.cpp

```diff
@@ -624,36 +1411,56 @@ std::wstring CUIforETWDlg::GenerateResultFilename() const
 	_strdate_s(date);
 	int hour, min, sec;
 	int year, month, day;
-#pragma warning(suppress : 4996)
-	const wchar_t* username = _wgetenv(L"USERNAME");
-	if (!username)
-		username = L"";
+
+	PCWSTR username_temp = NULL;
+	wchar_t usernameBuffer[MAX_PATH] = { 0 };
+	size_t usernameReturnValue = 0;
+	//[_wgetenv_s returns] Zero if successful.
+	const errno_t getUsernameResult = _wgetenv_s(&usernameReturnValue, usernameBuffer, L"USERNAME");
+	if (getUsernameResult == 0)
+		username_temp = usernameBuffer;
+	PCWSTR const username = username_temp;
+
+	//MAX_PATH is safe here, single file name component is limited to MAX_PATH;
+	//In fact, for NTFS, that's what MAX_PATH means!
+	//FAT is a different story.
+	//Anything else is wrong!
 	wchar_t fileName[MAX_PATH];
-	// Hilarious /analyze warning on this line from bug in _strtime_s annotation!
-	// warning C6054: String 'time' might not be zero-terminated.
-#pragma warning(suppress : 6054)
-	if (3 == sscanf_s(time, "%d:%d:%d", &hour, &min, &sec) &&
-		3 == sscanf_s(date, "%d/%d/%d", &month, &day, &year))
+
+//Eww.
+#pragma warning(suppress: 6054)
+	const bool validTimeStr = (3 == sscanf_s(time, "%d:%d:%d", &hour, &min, &sec));
+#pragma warning(suppress: 6054)
+	const bool validDateStr = (3 == sscanf_s(date, "%d/%d/%d", &month, &day, &year));
+
+	std::wstring filePart;
+	if (validTimeStr && validDateStr)
 	{
 		// The filenames are chosen to sort by date, with username as the LSB.
-		swprintf_s(fileName, L"%04d-%02d-%02d_%02d-%02d-%02d_%s", year + 2000, month, day, hour, min, sec, username);
+		const int res = swprintf_s(fileName, L"%04d-%02d-%02d_%02d-%02d-%02d_%s", year + 2000, month, day, hour, min, sec, username);

```

Comment:
Don't handle impossible errors. swprintf_s *will not* return an error code. It will fail if the text won't fit.